kratos/includes/gid_gauss_point_container.h

```diff
@@ -115,6 +115,8 @@ class GidGaussPointsContainer
                         GiD_fWriteScalar( ResultFile, it->Id(), ValuesOnIntPoint[index] );
                     }                    
                 }
+				// Set first entry to NaN for test below
+				ValuesOnIntPoint[0] = std::numeric_limits<double>::quiet_NaN();
```

Comment:
@pooyan-dadvand and @RiccardoRossi this is a bug in the existing code:
If no new results are retrieved from the conditions, all results were overwritten with the last result from the elements
This is there to prevent this from happening, but there is probably a nicer way

Reply:
nasty but I don't have better solution without changing the API.... 😞

Reply:
So shall I leave and merge it?
Or do you have a better solution?